### PR TITLE
optimize: support for more operational commands in seata-server.sh

### PR DESCRIPTION
**Pull request by doubleDimple**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
Support for more operational commands in seata-server.sh
start  ./seata-server.sh start 
stop   ./seata-server.sh stop
restart  ./seata-server.sh restart

### Ⅱ. Does this pull request fix one issue?
fixes https://github.com/seata/seata/issues/5522

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
The local test results are as follows
Test1: ./seata-server.sh 
image
Test2: ./seata-server.sh start
image
Test3: ./seata-server.sh start -p 9870
image
image
Test4: ./seata-server.sh stop
image
Test5: ./seata-server.sh restart
image
image
image
Test6: ./seata-server.sh --help
image
Test7: ./seata-server.sh -x
image

### CODE REVIEW

**Comment by doubleDimple** [comment]
1: ./seata-server.sh start
image
2: ./seata-server.sh stop 
image

3:./seata-server.sh restart
image

**Comment by xingfudeshi** [comment]
Please resolve the conflicting files.

**Comment by doubleDimple** [comment]
> Please resolve the conflicting files.
get

File: distribution/bin/seata-server.sh [hunk 1]
```diff
@@ -151,9 +151,62 @@ if [ ! -x "$BASEDIR"/logs ]; then
...
+        start_server
+        ;;
+    stop)
+        stop_server
+        ;;
+    restart)
+        restart_server
+        ;;
+    *)
+        echo "Usage: $0 {start|stop|restart}"
```
Comment (slievrly) [inline distribution/bin/seata-server.sh:207]: https://github.com/seata/seata/blob/83493d9e08bdb0e3512ba2121ee6309fd37d9af3/distribution/Dockerfile#L42

Comment (doubleDimple) [reply distribution/bin/seata-server.sh]: get

File: distribution/bin/seata-server.sh [hunk 2]
```diff
@@ -151,9 +151,61 @@ if [ ! -x "$BASEDIR"/logs ]; then
...
 
-# start
-echo "$JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS}" > ${BASEDIR}/logs/start.out 2>&1 &
-nohup $JAVACMD ${JAVA_OPT} ${CMD_LINE_ARGS} >> ${BASEDIR}/logs/start.out 2>&1 &
-echo "seata-server is starting, you can check the ${BASEDIR}/logs/start.out"
\ No newline at end of file
+start_server() {
+
+  echo "$JAVACMD ${JAVA_OPT}" > ${BASEDIR}/logs/start.out 2>&1 &
+  nohup $JAVACMD ${JAVA_OPT} >> ${BASEDIR}/logs/start.out 2>&1 &
```
Comment (slievrly) [inline distribution/bin/seata-server.sh:158]: should support external parameters

Comment (doubleDimple) [reply distribution/bin/seata-server.sh:158]: 收到，这里要不要在执行脚本的时候就检查下参数，如果参数错误或者输入help，直接提示help显示信息呐？

**Comment by slievrly** [comment]
@doubleDimple Pls resolve conflicts.

File: distribution/bin/seata-server.sh [hunk 3]
```diff
@@ -15,14 +15,399 @@
...
+  link=`expr "$ls" : '.*-> \(.*\)$'`
+  if expr "$link" : '/.*' > /dev/null; then
+    PRG="$link"
+  else
+    PRG=`dirname "$PRG"`/"$link"
+  fi
+done
+
+PRGDIR=`dirname "$PRG"`
+BASEDIR=`cd "$PRGDIR/.." >/dev/null; pwd`
```
Comment (a364176773) [inline distribution/bin/seata-server.sh:32]: 这一段在seata-setup.sh已经有了吧？

**Review by a364176773 — COMMENTED** [review]
冲突解决的有问题，把seata-setup.sh里的代码挪回来了

File: distribution/bin/seata-server.sh (hunk 3 above)
Comment (doubleDimple) [reply distribution/bin/seata-server.sh]: get

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**